### PR TITLE
[DBInstance] Revert isEngineMutable check to use previous model for same engine check

### PR DESCRIPTION
**Pull request by elnaggar-amr**
*Issue #, if available:*

*Description of changes:*
Revert isEngineMutable check to use previous model for same engine check

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java [hunk 1]
```diff
@@ -599,7 +609,6 @@ public static ModifyDbInstanceRequest modifyDbInstanceAfterCreateRequest(final R
                 .engineVersion(model.getEngineVersion())
                 .manageMasterUserPassword(model.getManageMasterUserPassword())
                 .masterUserPassword(model.getMasterUserPassword())
-                .masterUserPassword(model.getMasterUserPassword())
```
Comment (moataz-mhmd) [inline aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java:602]: Can you elaborate why `masterUserPassword` is being removed here?

File: aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java [hunk 2]
```diff
@@ -614,6 +623,11 @@ public static ModifyDbInstanceRequest modifyDbInstanceAfterCreateRequest(final R
             builder.storageThroughput(model.getStorageThroughput());
             builder.storageType(model.getStorageType());
         }
+
+        if (ResourceModelHelper.isOracleCDBEngine(model.getEngine())) {
+            builder.engine(model.getEngine());
```
Comment (moataz-mhmd) [inline aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java:628]: Is this parameter idempotent? What I mean is, if the instance being restored already had the CDB engine, and the Modify call includes the same CDB engine, will it succeed on the RDS side without making any changes or will it fail with a validation error? We have seen cases of non-idempotent behavior in the past, so we need to take that into consideration.

File: aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java (hunk 1 above)
Comment (elnaggar-amr) [reply aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java:602]: it's duplicated on line 611. not sure it was duplicate.

File: aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java (hunk 2 above)
Comment (elnaggar-amr) [reply aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java:628]: it's idempotent, I tried that manually. If there are no other changes though, there will be no modify after create as CDB to CDB is not conversion and will be skipped from CFN side.

File: aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java (hunk 1 above)
Comment (moataz-mhmd) [reply aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java:602]: Apology missed it :)

File: aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java [hunk 3]
```diff
@@ -599,7 +609,6 @@ public static ModifyDbInstanceRequest modifyDbInstanceAfterCreateRequest(final R
                 .engineVersion(model.getEngineVersion())
```
Comment (moataz-mhmd) [inline aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java:609]: From: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/oracle-cdb-converting.html
> When you upgrade your database engine version, you can't change the database architecture in the same operation. Therefore, to upgrade an Oracle Database 19c non-CDB to an Oracle Database 21c CDB, you first need to convert the non-CDB to a CDB in one step, and then upgrade the 19c CDB to a 21c CDB in a separate step.

I think if we try to do the CDB conversion and engine version upgrade from 19c to 21c at the same time, it's gonna be problematic on CFN handler. Can you please confirm if we're good to go with this scenario for both restoring and updating?

**Review by moataz-mhmd — COMMENTED** [review]
Make sure you take care of the rollback situation when making one-way changes, like CDB conversion. Let me give you an example to explain how we handle rollback during modifications. You can check out AllocatedStorage parameter case here: [GitHub Link](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/blob/master/aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java#L486-L489)

**Review by moataz-mhmd — APPROVED** [review]

**Review by dbbh — APPROVED** [review]

**Pull request merged**